src/PowerFlowData.jl

```diff
@@ -1,20 +1,111 @@
+        vector containing the indeces related to those buses that are not slack
+        ones.
+- `power_network_matrix::M`:
+        matrix used for the evaluation of either the power flows or bus angles, 
+        depending on the method considered.
+- `aux_network_matrix::N`:
+        matrix used for the evaluation of either the power flows or bus angles, 
+        depending on the method considered.
+
+"""
+struct PowerFlowData{M <: PNM.PowerNetworkMatrix, N, S <: Union{String, Char}}
     bus_lookup::Dict{Int, Int}
     branch_lookup::Dict{String, Int}
-    bus_activepower_injection::Vector{Float64}
-    bus_reactivepower_injection::Vector{Float64}
-    bus_activepower_withdrawals::Vector{Float64}
-    bus_reactivepower_withdrawals::Vector{Float64}
+    bus_activepower_injection::Matrix{Float64}
+    bus_reactivepower_injection::Matrix{Float64}
+    bus_activepower_withdrawals::Matrix{Float64}
+    bus_reactivepower_withdrawals::Matrix{Float64}
     bus_type::Vector{PSY.BusTypes}
-    bus_magnitude::Vector{Float64}
-    bus_angles::Vector{Float64}
-    branch_flow_values::Vector{Float64}
+    bus_magnitude::Matrix{Float64}
+    bus_angles::Matrix{Float64}
+    branch_flow_values::Matrix{Float64}
+    timestep_map::Dict{Int, S}
     valid_ix::Vector{Int}
     power_network_matrix::M
     aux_network_matrix::N
 end
 
-function PowerFlowData(::ACPowerFlow, sys::PSY.System)
+# AC Power Flow Data
+# TODO -> MULTI PERIOD: AC Power Flow Data
+"""
+Function for the definition of the PowerFlowData strucure given the System
+data, number of time periods to consider and their names.
+Calling this function will not evaluate the power flows and angles.
+NOTE: use it for AC power flow computations.
+
+# Arguments:
+- `::ACPowerFlow`:
+        use ACPowerFlow() to evaluate the AC OPF.
+- `sys::PSY.System`:
+        container storing the system data to consider in the PowerFlowData
+        structure.
+- `timesteps::Int`:
+        number of time periods to consider in the PowerFlowData structure. It 
+        defines the number of columns of the matrices used to store data.
+        Default value = 1.
+- `timestep_names::Vector{String}`:
+        names of the time periods defines by the argmunet "timesteps". Default
+        value = String[].
+
+WARNING: functions for the evaluation of the multi-period AC OPF still to be implemented.
+"""
+function PowerFlowData(
+    ::ACPowerFlow,
+    sys::PSY.System,
+    timesteps::Int = 1,
+    timestep_names::Vector{String} = String[])
+
+    # assign timestep_names
+    # timestep names are then allocated in a dictionary to map matrix columns
+    if timesteps != 0
+        if length(timestep_names) == 0
+            timestep_names = [string(i) for i in 1:timesteps]
+        elseif length(timestep_names) != timesteps
+            error("timestep_names field must have same length as timesteps")
+        end
+    end
+
+    # get data for calculations
     power_network_matrix = PNM.Ybus(sys)
```

Comment:
We should add connectivity checks here

Reply:
This is how I found it. Do you want to add the connectivity check as for the DC case with the ABA and BA matrix? Or do you refer to something different?